From ae03dda53cff7a67cde17f910d0ac7dbac1d8a98 Mon Sep 17 00:00:00 2001
From: Oliver Eftevaag <oliver.eftevaag@qt.io>
Date: Wed, 24 Sep 2025 14:22:10 +0200
Subject: [PATCH 091/239] ColorDialog: Comply with accessibility insights for
 Windows

Accessibility Insights for Windows is a Windows application that will
check the UI accessibility tree for any potential issues or
anti-patterns.

The eye dropper button, as well as the sliders, didn't have a name,
which was the main thing that the tool complained about.

Pick-to: 6.8
Change-Id: Ibe9d9e788d86fe3f46261221da12c1a4be5d220e
Reviewed-by: Mitch Curtis <mitch.curtis@qt.io>
(cherry picked from commit a21a49210f2c8f4af54f70eb455b4c4c67db864c)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 .../quickdialogsquickimpl/qml/+Fusion/ColorDialog.qml       | 6 ++++++
 .../quickdialogsquickimpl/qml/+Imagine/ColorDialog.qml      | 6 ++++++
 .../quickdialogsquickimpl/qml/+Material/ColorDialog.qml     | 6 ++++++
 .../quickdialogsquickimpl/qml/+Universal/ColorDialog.qml    | 6 ++++++
 src/quickdialogs/quickdialogsquickimpl/qml/ColorDialog.qml  | 6 ++++++
 5 files changed, 30 insertions(+)

diff --git a/src/quickdialogs/quickdialogsquickimpl/qml/+Fusion/ColorDialog.qml b/src/quickdialogs/quickdialogsquickimpl/qml/+Fusion/ColorDialog.qml
index 1bab4d5b1b..891151971a 100644
--- a/src/quickdialogs/quickdialogsquickimpl/qml/+Fusion/ColorDialog.qml
+++ b/src/quickdialogs/quickdialogsquickimpl/qml/+Fusion/ColorDialog.qml
@@ -78,6 +78,8 @@ ColorDialogImpl {
             Layout.preferredWidth: implicitHeight
             Layout.alignment: Qt.AlignRight
             Layout.rightMargin: 6
+
+            Accessible.name: qsTr("Eye dropper")
         }
     }
 
@@ -126,6 +128,8 @@ ColorDialogImpl {
                 }
             }
 
+            Accessible.name: qsTr("Hue")
+
             Layout.fillWidth: true
             Layout.leftMargin: 12
             Layout.rightMargin: 12
@@ -180,6 +184,8 @@ ColorDialogImpl {
                 }
             }
 
+            Accessible.name: qsTr("Alpha")
+
             Layout.fillWidth: true
             Layout.leftMargin: 12
             Layout.rightMargin: 12
diff --git a/src/quickdialogs/quickdialogsquickimpl/qml/+Imagine/ColorDialog.qml b/src/quickdialogs/quickdialogsquickimpl/qml/+Imagine/ColorDialog.qml
index 16c9ed9fbd..8096a706a6 100644
--- a/src/quickdialogs/quickdialogsquickimpl/qml/+Imagine/ColorDialog.qml
+++ b/src/quickdialogs/quickdialogsquickimpl/qml/+Imagine/ColorDialog.qml
@@ -82,6 +82,8 @@ ColorDialogImpl {
 
             Layout.alignment: Qt.AlignRight
             Layout.rightMargin: 16
+
+            Accessible.name: qsTr("Eye dropper")
         }
     }
 
@@ -102,6 +104,8 @@ ColorDialogImpl {
             value: control.hue
             onMoved: function() { control.hue = value; }
 
+            Accessible.name: qsTr("Hue")
+
             Layout.fillWidth: true
             Layout.leftMargin: 16
             Layout.rightMargin: 16
@@ -189,6 +193,8 @@ ColorDialogImpl {
                 }
             }
 
+            Accessible.name: qsTr("Alpha")
+
             Layout.fillWidth: true
             Layout.leftMargin: 16
             Layout.rightMargin: 16
diff --git a/src/quickdialogs/quickdialogsquickimpl/qml/+Material/ColorDialog.qml b/src/quickdialogs/quickdialogsquickimpl/qml/+Material/ColorDialog.qml
index 6d7e4991e8..fc7d3613ef 100644
--- a/src/quickdialogs/quickdialogsquickimpl/qml/+Material/ColorDialog.qml
+++ b/src/quickdialogs/quickdialogsquickimpl/qml/+Material/ColorDialog.qml
@@ -75,6 +75,8 @@ ColorDialogImpl {
 
             Layout.alignment: Qt.AlignRight
             Layout.rightMargin: 24
+
+            Accessible.name: qsTr("Eye dropper")
         }
     }
 
@@ -123,6 +125,8 @@ ColorDialogImpl {
                 }
             }
 
+            Accessible.name: qsTr("Hue")
+
             Layout.fillWidth: true
             Layout.leftMargin: 12
             Layout.rightMargin: 12
@@ -180,6 +184,8 @@ ColorDialogImpl {
                 }
             }
 
+            Accessible.name: qsTr("Alpha")
+
             Layout.fillWidth: true
             Layout.leftMargin: 12
             Layout.rightMargin: 12
diff --git a/src/quickdialogs/quickdialogsquickimpl/qml/+Universal/ColorDialog.qml b/src/quickdialogs/quickdialogsquickimpl/qml/+Universal/ColorDialog.qml
index 54f772895d..b2b2a6492e 100644
--- a/src/quickdialogs/quickdialogsquickimpl/qml/+Universal/ColorDialog.qml
+++ b/src/quickdialogs/quickdialogsquickimpl/qml/+Universal/ColorDialog.qml
@@ -78,6 +78,8 @@ ColorDialogImpl {
 
             Layout.alignment: Qt.AlignRight
             Layout.rightMargin: 18
+
+            Accessible.name: qsTr("Eye dropper")
         }
     }
 
@@ -126,6 +128,8 @@ ColorDialogImpl {
                 }
             }
 
+            Accessible.name: qsTr("Hue")
+
             Layout.fillWidth: true
             Layout.leftMargin: 12
             Layout.rightMargin: 12
@@ -183,6 +187,8 @@ ColorDialogImpl {
                 }
             }
 
+            Accessible.name: qsTr("Alpha")
+
             Layout.fillWidth: true
             Layout.leftMargin: 12
             Layout.rightMargin: 12
diff --git a/src/quickdialogs/quickdialogsquickimpl/qml/ColorDialog.qml b/src/quickdialogs/quickdialogsquickimpl/qml/ColorDialog.qml
index 432d04a387..7c8fab04f8 100644
--- a/src/quickdialogs/quickdialogsquickimpl/qml/ColorDialog.qml
+++ b/src/quickdialogs/quickdialogsquickimpl/qml/ColorDialog.qml
@@ -75,6 +75,8 @@ ColorDialogImpl {
                 Layout.preferredWidth: implicitHeight
                 Layout.alignment: Qt.AlignRight
                 Layout.rightMargin: 6
+
+                Accessible.name: qsTr("Eye dropper")
             }
         }
     }
@@ -124,6 +126,8 @@ ColorDialogImpl {
                 }
             }
 
+            Accessible.name: qsTr("Hue")
+
             Layout.fillWidth: true
             Layout.leftMargin: 12
             Layout.rightMargin: 12
@@ -181,6 +185,8 @@ ColorDialogImpl {
                 }
             }
 
+            Accessible.name: qsTr("Alpha")
+
             Layout.fillWidth: true
             Layout.leftMargin: 12
             Layout.rightMargin: 12
-- 
2.51.2

